php/example_code/aws_utilities/AWSServiceClass.php

```diff
@@ -43,4 +44,68 @@ public function customWaiter($function, $verbose = false)
         }
         return $result;
     }
+
+    /**
+     * This is a very common task for all services, so this is a helper function to use everywhere.
+     */
+
+    /**
+     * @param $roleName string the name to give the role
+     * @param string|null $rolePolicyDocument
+     * @param $clientArgs array|null optional array of client arguments
+     * @param $verbose bool whether to echo debug information
+     * @return string the created role's ARN
+     */
+    public function createRole(
+        string $roleName,
+        string $rolePolicyDocument = null,
+        array $clientArgs = null,
+        bool $verbose = false
+    ): string {
+        $clientArgs = $clientArgs ?: [
+            'region' => 'us-west-2',
+            'version' => 'latest',
+            'profile' => 'default',
+        ];
+        $iamService = new IAMService($clientArgs);
+
+        $rolePolicyDocument = $rolePolicyDocument ?: "{
+            \"Version\": \"2012-10-17\",
+            \"Statement\": [
+                {
+                    \"Effect\": \"Allow\",
+                    \"Principal\": {
+                        \"Service\": \"states.amazonaws.com\"
```

Comment:
Do you want to lock this to Step Functions or would it be better to parameterize the service principal?

Reply:
Yes...but later